session/session.go

```diff
@@ -478,9 +486,16 @@ func (s *session) doCommit(ctx context.Context) error {
 	if !s.txn.Valid() {
 		return nil
 	}
+
+	// to avoid session set overlap the txn set.
+	if s.GetDiskFullOpt() != kvrpcpb.DiskFullOpt_NotAllowedOnFull {
```

Comment:
But in this way, the txn setting overwrites the session setting. To do it inversely, you need to set the txn setting to the session setting when the transaction begins.

Reply:
There will be different entrances for transaction settings. The main entrance is to directly set txn, but there are also cases where txn is set through session. Therefore, it is necessary to choose the setting that is not DiskFullOpt_NotAllowedOnFull among the two.